Makefile

```diff
@@ -14,72 +16,72 @@ prow-crd:
 	# HMAC_TOKEN is used for encrypting Github webhook payloads.
-	oc create secret generic hmac-token --from-literal=hmac=${HMAC_TOKEN} -o yaml --dry-run | oc apply -n ci -f -
+	oc create secret generic hmac-token --from-literal=hmac=${HMAC_TOKEN} -o yaml --dry-run | oc apply -n $(CI_NAMESPACE) -f -
 	# OAUTH_TOKEN is used for manipulating Github PRs/issues (labels, comments, etc.).
-	oc create secret generic oauth-token --from-literal=oauth=${OAUTH_TOKEN} -o yaml --dry-run | oc apply -n ci -f -
+	oc create secret generic oauth-token --from-literal=oauth=${OAUTH_TOKEN} -o yaml --dry-run | oc apply -n $(CI_NAMESPACE) -f -
 .PHONY: prow-secrets
 
 prow-images:
-	oc process -f cluster/ci/config/prow/prow_images.yaml | oc apply -f -
+	oc process -f cluster/ci/config/prow/prow_images.yaml -p NAMESPACE=$(CI_NAMESPACE) | oc apply -f -
 .PHONY: prow-images
 
 prow-rbac:
-	oc process -f cluster/ci/config/prow/openshift/deck_rbac.yaml | oc apply -f -
-	oc process -f cluster/ci/config/prow/openshift/hook_rbac.yaml | oc apply -f -
-	oc process -f cluster/ci/config/prow/openshift/horologium_rbac.yaml | oc apply -f -
-	oc process -f cluster/ci/config/prow/openshift/jenkins_operator_rbac.yaml | oc apply -f -
-	oc process -f cluster/ci/config/prow/openshift/plank_rbac.yaml | oc apply -f -
-	oc process -f cluster/ci/config/prow/openshift/sinker_rbac.yaml | oc apply -f -
-	oc process -f cluster/ci/config/prow/openshift/splice_rbac.yaml | oc apply -f -
+	oc process -f cluster/ci/config/prow/openshift/deck_rbac.yaml -p NAMESPACE=$(CI_NAMESPACE) | oc apply -f -
+	oc process -f cluster/ci/config/prow/openshift/hook_rbac.yaml -p NAMESPACE=$(CI_NAMESPACE) | oc apply -f -
+	oc process -f cluster/ci/config/prow/openshift/horologium_rbac.yaml -p NAMESPACE=$(CI_NAMESPACE) | oc apply -f -
+	oc process -f cluster/ci/config/prow/openshift/jenkins_operator_rbac.yaml -p NAMESPACE=$(CI_NAMESPACE) | oc apply -f -
+	oc process -f cluster/ci/config/prow/openshift/plank_rbac.yaml -p NAMESPACE=$(CI_NAMESPACE) | oc apply -f -
+	oc process -f cluster/ci/config/prow/openshift/sinker_rbac.yaml -p NAMESPACE=$(CI_NAMESPACE) | oc apply -f -
+	oc process -f cluster/ci/config/prow/openshift/splice_rbac.yaml -p NAMESPACE=$(CI_NAMESPACE) | oc apply -f -
 
 prow-services:
-	oc process -f cluster/ci/config/prow/openshift/deck.yaml | oc apply -f -
-	oc process -f cluster/ci/config/prow/openshift/hook.yaml | oc apply -f -
-	oc process -f cluster/ci/config/prow/openshift/horologium.yaml | oc apply -f -
-	oc process -f cluster/ci/config/prow/openshift/jenkins_operator.yaml | oc apply -f -
-	oc process -f cluster/ci/config/prow/openshift/jenkins_proxy.yaml | oc apply -f -
-	oc process -f cluster/ci/config/prow/openshift/plank.yaml | oc apply -f -
-	oc process -f cluster/ci/config/prow/openshift/sinker.yaml | oc apply -f -
-	oc process -f cluster/ci/config/prow/openshift/splice.yaml | oc apply -f -
+	oc process -f cluster/ci/config/prow/openshift/deck.yaml -p NAMESPACE=$(CI_NAMESPACE) | oc apply -f -
```

Comment:
This is fishy. @csrwng do you have any opinions on it?

Reply:
What do you mean?

Reply:
I feel like having to specify the namespace on every object in a template is mis-use of a template

Reply:
Looking at the templates, it seems that we could leave out the NAMESPACE parameter completely. @kargakis is there a specific case where the namespace must be specified? I think leaving it out of the template will make it more portable and easier to instantiate

Reply:
The only reason is to have a preset default so we wont depend on the context. Agreed that it makes the templates less portable. I am going to remove it.

Reply:
Another reason for parameterizing the namespace I just remembered is that we are using clusterroles and they require the namespace of the service account to be present inside a clusterrolebinding. We don't really need clusterroles (https://github.com/openshift/release/issues/125) so I went ahead and switched them to roles.

Reply:
I want to use sinker for garbage collecting pods in the origin-ci namespace. Meaning sinker needs to use a clusterrole. How are we going to infer the namespace inside the template in that case?

Reply:
so I've added cleanup to the pipelines for unit tests and verification tests ... but builds are not yet getting cleaned up. I think that can be done with the regular build pruner though.